File: 49.c
#include <stdio.h>
#include <string.h>
#include <math.h>

int main() {
    int t;
    scanf("%d", &t);

    while (t--) {
        char num[1001];
        scanf("%s", num);

        int length = strlen(num);
        int count = 0;
        int results[100];  // To store the place values

        for (int i = 0; i < length; i++) {
            int digit = num[i] - '0';
            if (digit != 0) {
                int power = length - 1 - i;
                results[count++] = digit * pow(10, power);
            }
        }

        printf("%d\n", count);
        for (int i = 0; i < count; i++) {
            printf("%d ", results[i]);
        }
        printf("\n");
    }

    return 0;
}

File: 49.py
t = int(input())

for _ in range(t):
    n = input()
    l = []
    length = len(n)
    for i in range(length):
        digit = int(n[i])
        power = ((length-1)-i)
        place_value = digit * (10 ** power)
        if digit != 0:
            l.append(place_value)
    print(len(l))
    for items in l:
        print(items, end=" ")
    print()
